Source/JavaScriptCore/offlineasm/generate_offset_extractor.rb

```diff
@@ -96,7 +97,7 @@ def emitMagicNumber
     depfile.puts(Shellwords.join(sources.sort))
 end
 
-File.open(outputFlnm, "w") {
+Tempfile.create([File.basename(outputFlnm, ".h"), ".h"], File.dirname(outputFlnm)) {
```

Comment:
Will this temporary file be stuck there until deleting the build directory if the script gets interrupted? Trying to think about whether this would cause problems.

Also, are there any potential concerns with creating files that the build system doesn't know about in the build directory?

Reply:
It can do nothing when it receives SIGKILL, which is what I'd expect to happen on the bots under certain circumstances. Curious about whether it at least deletes temporary files on SIGTERM.

Reply:
Yeah, it deletes on SIGTERM.

> Also, are there any potential concerns with creating files that the build system doesn't know about in the build directory?
XCBuild will ignore it, since it only tracks and prunes files in its build description. (I would assume ninja is the same, as it doesn't have any special stale file detection.)

It may be better to use a static name for the temporary file, so after a SIGKILL, the next rebuild would overwrite the partially-written output file, instead of leaving it behind forever.